### PR TITLE
Fix AllRooms #1316

### PR DESCRIPTION
**Pull request by Pertex**
Fix AllRooms #1316
moving editor types to core.aslx
changing the logic from "you are a room" to "you are NOT a room"

### CODE REVIEW

**Comment by Pertex** [comment]
did I say I hate GIT? I dont know how to get rid of that *§%&#! datetime commit, so could you just check the files of Fix AllRooms #1316?

**Comment by KVonGit** [comment]
> did I say I hate GIT? I dont know how to get rid of that *§%&#! datetime commit, so could you just check the files of Fix AllRooms #1316?

We have to start from a fresh branch for each new PR.

1 - Go to your forks main page on GitHub

2 - Be sure it says your master branch is up to date with `textadventures/quest/:master`

image

3 - Create a new branch in GitHub Desktop (or in VS), and be sure it is based on the `master` branch

4 - Make your changes and do the PR from that new branch

---
Since it's so close to midnight in your neck of the woods, I can to this in a PR for you if you like, unless you'd rather do it,

**Comment by KVonGit** [comment]
Oh, you'll need to sync your fork before submitting any new ones, too, or it will possibly revert changes committed since you last synced your fork.

image

---
I don't have it in my screenshot of your branch, but you will see the button: image

---
WARNING: I don't know enough about GitHub to know what will happen with your 4 commits that are ahead. So, you might want to wait until you've finished anything you currently have open before you sync.

File: WorldModel/WorldModel/Core/CoreFunctions.aslx [hunk 1]
```diff
@@ -568,10 +568,14 @@
     return (l)
   </function>
   
-  <!-- Added by KV -->
-
   <function name="AllRooms" type="objectlist">
-    return (FilterByAttribute(AllObjects(),"isroom",true))
+    l = NewObjectList()
+    foreach (o, AllObjects()) {
```
Comment (KVonGit) [inline WorldModel/WorldModel/Core/CoreFunctions.aslx:573]: This `foreach` is the same thing (or *should* be the same thing) as `FilterByAttribute(AllObjects(),"isroom",true)`

https://github.com/textadventures/quest/blob/8625f9b071acee23efb18514ae68b398de722f29/WorldModel/WorldModel/Core/CoreFunctions.aslx#L532-L542

---
Did it not return the same list the way it was? (Just curious.)

**Comment by KVonGit** [comment]
Closing this to avoid confusion.

I put all these changes in #1326

File: WorldModel/WorldModel/Core/CoreFunctions.aslx (hunk 1 above)
Comment (Pertex) [reply WorldModel/WorldModel/Core/CoreFunctions.aslx:573]: no, I have to add the "is object and room" which does not have the attribute isroom

Comment (KVonGit) [reply WorldModel/WorldModel/Core/CoreFunctions.aslx:573]: So... Why does mine work? I don't understand that.

The only thing I see different is I added `isroom` at the end of the `defaultobject` type's attributes, and you have yours at the very beginning.

YOURS:
image

---

MINE:
image

---
And our `default_object` and `default_room` bits are inverted:

YOURS:
image

---

MINE:
image

---
I didn't do any of that on purpose, either.

**Pull request closed without merging**